caikit_nlp/modules/text_generation/peft_prompt_tuning.py

```diff
@@ -465,20 +486,40 @@ def train(
                 silence_progress_bars=silence_progress_bars,
                 # NOTE: following can override above arguments in order
                 **filtered_training_arguments,
+                **processing_configuration,
             )
 
-            base_model_trainer = base_model.get_trainer(
-                train_dataset=training_dataset, model=peft_model, **training_args
-            )
+            if torch.cuda.is_available():
+                # NOTE: torch distributed can hang if run on CPUs,
+                # to avoid that, specially for unit tests, we are only
+                # running below when GPUs are available
+                launch_config = get_torch_elastic_launch_config(
```

Comment:
We might want to revisit this - right now it is mirroring fine tuning, but do we want to use torch launch when there's only one device?

Reply:
yeah, we can do some testing and make it to work only with 1 GPU 🤔